edb/lang/edgeql/ast.py

```diff
@@ -235,45 +244,66 @@ class UnaryOp(Expr):
     operand: Expr
 
 
-class Ptr(Base):
-    ptr: ObjectRef
-    direction: str
-    target: Expr
-    type: str
+class TypeExpr(Base):
+    pass
 
 
-class _TypeName(Expr):
-    pass
+class TypeOf(TypeExpr):
+    expr: Expr
 
 
-class TypeName(_TypeName):
-    name: str  # name is used for types in named tuples
+class TypeName(TypeExpr):
+    name: str  # XXX: name is used for types in named tuples
```

Comment:
Why `XXX`?

Reply:
Oh, that's right, I wanted to come back and fix it in the same fashion as a couple of other things in the parser. It should be a named tuple for parser use, rather than weird reuse of the TypeName.

Reply:
Hm, I thought it was something else, nevermind. The comment is accurate and the `XXX` is not necessary.